src/modules/blockchainApplication/connection/components/ConnectionSummary/AccountsSelector.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import AccountRow from '@account/components/AccountRow';
+import CheckBox from '@theme/CheckBox';
+import styles from './connectionSummary.css';
+
+const AccountsSelector = ({ setAddresses, addresses, accounts }) => {
+  const [selectAll, setSelectAll] = useState(false);
+  const { t } = useTranslation();
+
+  const onSelect = (e) => {
+    setSelectAll(false);
+    if (e.target.checked) {
+      setAddresses([...addresses, e.target.name]);
+    } else {
+      setAddresses(addresses.filter(item => item !== e.target.name));
+    }
+  };
+
+  const onSelectAll = () => {
+    if (selectAll) {
+      setAddresses([]);
+    } else {
+      setAddresses(accounts.map(item => item.metadata.address));
+    }
+    setSelectAll(!selectAll);
```

Comment:
```suggestion
    setSelectAll((state) => !state);
```

Reply:
This is already changes and has a different implememntation